Extension/src/LanguageServer/customProviders.ts

```diff
@@ -0,0 +1,185 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+
+import { CustomConfigurationProvider, Version, SourceFileConfigurationItem } from 'vscode-cpptools';
+import * as vscode from 'vscode';
+
+/**
+ * An interface that is guaranteed to be backward compatible with version 0
+ */
+export interface CustomConfigurationProvider1 extends CustomConfigurationProvider {
+    isValid: boolean;
+    version: Version;
+}
+
+/**
+ * Wraps the incoming CustomConfigurationProvider so that we can treat all of them as if they were the same version (e.g. latest)
+ */
+class CustomProviderWrapper implements CustomConfigurationProvider1 {
+    private provider: CustomConfigurationProvider;
+    private _version: Version;
+
+    constructor(provider: CustomConfigurationProvider, version: Version) {
+        this.provider = provider;
+        if (provider.extensionId && version === Version.v0) {
+            version = Version.v1;   // provider implemented the new API but is interfacing with the extension using the old API version.
```

Comment:
Capital "Provider". Also, what's up with multiple spaces before //? Can we stick to 1 space? I haven't seen multiple spaces before // used anywhere else.